hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/metrics/FederationRPCPerformanceMonitor.java

```diff
@@ -153,7 +157,7 @@ public void proxyOpComplete(boolean success, String nsId,
     if (success) {
       long proxyTime = getProxyTime();
       if (proxyTime >= 0) {
-        if (metrics != null) {
+        if (metrics != null && !nsId.equals(CONCURRENT)) {
```

Comment:
Can you switch the test to `!CONCURRENT.equals(nsId)`

Looking at the code, the one case where nsId is null, success is also equal to false so we don't get a nullpointerexception. However, it is still good to switch the code to be more defensive.